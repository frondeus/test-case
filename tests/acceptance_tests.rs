#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
