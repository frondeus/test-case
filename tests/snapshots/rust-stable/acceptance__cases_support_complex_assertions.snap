---
source: tests/acceptance_tests.rs
assertion_line: 106
expression: output
---
    empty::vec_0_expects_complex_empty
---- empty::vec_0_expects_complex_empty stdout ----
error: test failed, to rerun pass '--lib'
failures:
failures:
running 54 tests
test combinators::_0_3_expects_complex_gt_0_0_and_lt_1_0_or_gt_1_2 ... ok
test combinators::_0_7_expects_complex_gt_0_0_and_lt_1_0_or_gt_1_2 ... ok
test combinators::_1_0_expects_complex_gt_0_0_and_lt_5_0 ... ok
test combinators::_1_0_expects_complex_gt_0_0_or_lt_0_0 ... ok
test combinators::_1_0_expects_complex_gt_0_0_or_lt_1_5_and_lt_2_0 ... ok
test combinators::_2_0_expects_complex_gt_0_0_or_lt_0_0 ... ok
test combinators::_2_0_expects_complex_gt_0_0_or_lt_0_0_and_lt_1_0 ... ok
test combinators_with_arrays::vec_1_2_3_expects_complex_contains_1_and_contains_2_and_contains_in_order_2_3_ ... ok
test combinators_with_arrays::vec_1_2_3_expects_complex_contains_1_and_contains_3_or_contains_5 ... ok
test combinators_with_arrays::vec_1_2_3_expects_complex_contains_1_or_contains_4 ... ok
test combinators_with_arrays::vec_1_2_3_expects_complex_contains_1_or_contains_4_and_contains_2 ... ok
test combinators_with_arrays::vec_1_2_3_expects_complex_contains_6_and_contains_7_or_contains_1 ... ok
test combinators_with_arrays::vec_1_2_3_expects_complex_contains_6_and_contains_7_or_contains_1_and_contains_in_order_1_2_3_ ... ok
test complex_tests::almost_eq1 ... ok
test complex_tests::almost_eq2 ... ok
test complex_tests::eq1 ... ok
test complex_tests::eq2 ... ok
test complex_tests::geq1 ... ok
test complex_tests::geq2 ... ok
test complex_tests::gt1 ... ok
test complex_tests::gt2 ... ok
test complex_tests::leq1 ... ok
test complex_tests::leq2 ... ok
test complex_tests::lt1 ... ok
test complex_tests::lt2 ... ok
test contains_tests::vec_1_2_3_4_expects_complex_contains_1 ... ok
test contains_tests::vec_1_2_3_4_expects_complex_contains_in_order_3_4_ ... ok
test count_general::_abcd_chars_expects_complex_count_4 ... ok
test count_general::std_iter_once_2_expects_complex_count_1 ... ok
test count_general::vec_0_1_3_expects_complex_count_3 ... ok
test count_vec::vec_0_1_3_expects_complex_count_3 ... ok
test count_vec::vec_0_1_expects_complex_count_2 ... ok
test create_path::_cargo_toml_expects_complex_path_path ... ok
test create_path::_src_lib_rs_expects_complex_path_file ... ok
test create_path::long_dir ... ok
test create_path::short_dir ... ok
test empty::vec_0_expects_complex_empty ... FAILED
test empty::vec_expects_complex_empty ... ok
test in_parens::_2_0_expects_complex_eq_2_0 ... ok
test len_byte_str::b_ab_expects_complex_len_2 ... ok
test len_byte_str::b_abc_expects_complex_len_3 ... ok
test len_str::_ab_expects_complex_len_2 ... ok
test len_str::_abc_expects_complex_len_3 ... ok
test len_string::_ab_expects_complex_len_2 ... ok
test len_string::_abc_expects_complex_len_3 ... ok
test len_vec::vec_0_1_3_expects_complex_len_3 ... ok
test len_vec::vec_0_1_expects_complex_len_2 ... ok
test not_complex::_1_0_expects_complex_not_almost_2_1_p_0_01 ... ok
test not_complex::_1_0_expects_complex_not_eq_2_5 ... ok
test not_contains_tests::vec_1_2_3_4_expects_complex_not_contains_5 ... ok
test not_contains_tests::vec_1_2_3_4_expects_complex_not_contains_in_order_3_2_ ... ok
test not_path::_cargo_toml_parse_unwrap_expects_complex_not_path_dir ... ok
test not_path::_cargo_yaml_parse_unwrap_expects_complex_not_path_path ... ok
test not_path::_src_parse_unwrap_expects_complex_not_path_file ... ok
test result: FAILED. 53 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
thread 'empty::vec_0_expects_complex_empty' panicked at 'assertion failed: _result.is_empty()', src/lib.rs:115:1
